src/imitation/algorithms/preference_comparisons.py

```diff
@@ -1114,20 +1125,95 @@ def __init__(
                 on the fly by specifying an `epoch_multiplier` in `self.train()`
                 if longer training is desired in specific cases).
             lr: the learning rate
-            weight_decay: the weight decay factor for the reward model's weights
-                to use with ``th.optim.AdamW``. This is similar to but not equivalent
-                to L2 regularization, see https://arxiv.org/abs/1711.05101
+            seed: the random seed to use for splitting the dataset into training
+                and validation.
             custom_logger: Where to log to; if None (default), creates a new logger.
+            reg_class: the regularization class to use. If not specified, no
```

Comment:
Why have `preference_comparisons` construct the regularizer, rather than have the caller construct it and pass it in? Both conventions have their pros/cons. We've leaned towards the latter so far in this code (e.g. `model` and `loss` must both be constructed in caller), and by defauld I'd lean towards keeping it consistent unless there's a clear benefit to deviating.

Reply:
The reason this is not possible currently is because the regularizer takes a logger and an optimizer, which is not accessible outside the scope of the initializer. You could inject them later, but then you might as well pass the class and construct the instance in `__init__()`.

Reply:
I've created a factory class that takes care of all the input validation, let me know your thoughts.